### PR TITLE
runCopro.sh with fhenix as DA

### PR DESCRIPTION
**Pull request by eshelB**
runcopro now deploys a second fhenix chain to act as the DA, and deploys the ct-registry contract to that chain

### CODE REVIEW

File: http/runCopro.sh [hunk 1]
```diff
@@ -32,6 +33,10 @@ if [ ! -d "$FheosDir" ]; then
   missing_dirs+=("Fheos Directory: $FheosDir")
 fi
 
+if [ ! -d "$FheosDir" ]; then
```
Comment (haimbj1) [inline http/runCopro.sh:36]: Did you mean $CtRegistryDir
Should we enforce the script? or only in certain config?

File: http/runCopro.sh [hunk 2]
```diff
@@ -57,29 +62,52 @@ function success(){
...
+            err "Error: Failed to run Local Fhenix for consumer chain"
         fi 
-        success "Local Fhenix is started"
-        listenDockerLogs
+        success "Local Fhenix (consumer) is started"
+        listenDockerLogs $name
+    fi
+}
+
+function runningLocalFhenixSecondInstance {
```
Comment (haimbj1) [inline http/runCopro.sh:97]: I suggest edit the runningLocalFhenix  with parameters rather copy the functionality, if you pass name, port1, port2,port3  you can just call this twice with different inputs.

File: http/runCopro.sh [hunk 3]
```diff
@@ -117,12 +145,27 @@ function runningFheosServer {
...
     cd $TaskManagerDir
     pnpm pnpm clean
     output=$(pnpm task:deploy)
     if [ $? -ne 0 ]; then
-        err "Error: Failed to deploy contracts"
+        err "Error: Failed to deploy TaskManager contracts"
+    fi
+}
+
+function deployContractCtRegistry {
```
Comment (haimbj1) [inline http/runCopro.sh:162]: Same here, we can create a deployContract function that get directory and name, and then you can call it twice under `deployContracts` with TM / CtRegistry

File: http/runCopro.sh [hunk 4]
```diff
@@ -131,6 +174,11 @@ function copyDeployedTM {
     cp $TaskManagerDir/deployments/localfhenix/TaskManager.json $AggregatorDir
 }
 
+function copyDeployedCtRegistry {
```
Comment (haimbj1) [inline http/runCopro.sh:177]: Same here

**Review by haimbj1 — CHANGES_REQUESTED** [review]

**Comment by eshelB** [comment]
do not merge yet, still testing

**Comment by eshelB** [comment]
merge is possible now

File: http/runCopro.sh [hunk 5]
```diff
@@ -117,18 +137,21 @@ function runningFheosServer {
 }
 
 function deployContracts {
-    step "Running deploy TaskManager and Example contracts"
-    cd $TaskManagerDir
+    local contractRepo=$1
+
+    step "Running deploy $(basename "$contractRepo")"
+    cd $contractRepo
     pnpm pnpm clean
```
Comment (haimbj1) [inline http/runCopro.sh:144]: Not in your code, but is it intentional pnpm twice?

**Review by haimbj1 — APPROVED** [review]

File: http/runCopro.sh (hunk 5 above)
Comment (eshelB) [reply http/runCopro.sh:144]: probably not

**Comment by eshelB** [comment]
Actually I still find bugs, testing again

**Pull request merged**